### ISSUE TITLE
FR: Single node cluster (one master, and no nodes)

### ISSUE THREAD

**Issue by jedateach** [kubernetes/kops#2688]
It is not clear how to deploy a single node cluster, with a single, schedule-able master. This will be useful for developers getting started to keep costs down, or perhaps could help in testing scenarios. Minikube works for local dev, but there isn't a simple equivalent solution for live servers.

Issue #152 suggests that nodes size could be set to `0`: https://github.com/kubernetes/kops/issues/152

However, I have tried this from both in the cli:
```
kops create cluster --node-count=0 ${CLUSTER_NAME}
```
That didn't affect the maxSize and minSize values in the nodes ig file.

Values can be changed by editing the nodes instance group config:
```
kops edit ig nodes --name=${CLUSTER_NAME}
```
After editing however, the resulting cluster config output yaml appears correct, but running `kops update cluster ${CLUSTER_NAME}` will still include a single instance launch configuration for the `nodes` instance group.

**Comment by jedateach** [comment]
Should this work already, and we need some documentation, or is this a genuine feature request?

**Comment by jedateach** [comment]
I'm closing this, as I appear to have fixed the issue. Autoscaling group for nodes set min and max values to `0`.

**Comment by vinayagg** [comment]
Hi Jeremy,

Can you (or anyone) give little details how to do this? My questions are:
1. does --node-count=0 automatically works? I tried it but the resulting IG has node count=2. 
Do I need to manually edit node ig config with node count = 0 before starting cluster?

2. With this config, is master automatically configured to schedule pods? AFAIK, by default pods are not scheduled on master nodes.

**Comment by jedateach** [comment]
Hi @vinayagg,
1. No I couldn't get that to work myself. I had to do the manual edit.
2. Yes, I had to remove the noschedule taint from the master - which is conveniently provided by the `uncordon` command:

Remove noschedule taint from single master:
```sh
kubectl uncordon $MASTER_NODE_NAME
```

**Comment by jedateach** [comment]
Because the solution to this issue is really a workaround what kops can't do, I'll reopen the issue.

**Comment by vinayagg** [comment]
I had opened another issue, they are now duplicate
https://github.com/kubernetes/kops/issues/3073

**Comment by jedateach** [comment]
Ok, great thanks.
